comp/networkpath/npcollector/npcollectorimpl/pathteststore/pathteststore.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package pathteststore handle pathtest storage
+package pathteststore
+
+import (
+	"sync"
+	time "time"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/networkpath/npcollector/npcollectorimpl/common"
+)
+
+var timeNow = time.Now
+
+// PathtestContext contains Pathtest information and additional flush related data
+type PathtestContext struct {
+	Pathtest *common.Pathtest
+
+	nextRun           time.Time
+	runUntil          time.Time
+	lastFlushTime     time.Time
+	lastFlushInterval time.Duration
+}
+
+// LastFlushInterval returns last flush interval
+func (p *PathtestContext) LastFlushInterval() time.Duration {
+	return p.lastFlushInterval
+}
+
+// SetLastFlushInterval sets last flush interval
+func (p *PathtestContext) SetLastFlushInterval(lastFlushInterval time.Duration) {
+	p.lastFlushInterval = lastFlushInterval
+}
+
+// Store is used to accumulate aggregated contexts
+type Store struct {
+	logger log.Component
+
+	contexts map[uint64]*PathtestContext
```

Comment:
reviewing for posterity: I think we will want to impose a size limit on this map to keep memory under control in cases of unexpected numbers of pathtest contexts

Reply:
I'm not sure yet if there is a need for a configuration that limits the # of contexts, the pathtest contexts are quite light in memory, I think we might hit other resources limits (input chan buffer, process chan buffer, # concurrent traceroutes) before memory being an issue due to `map[uint64]*PathtestContext` being too large.

If the volume of NPM conns is very high, I think those are the symptoms:
- the interval of the pathtest will become higher, e.g. we expect 5min, but it actually runs every 8min
- input_chan_buffer becomes full and we start dropping new conns

I think adding a limit to `map[uint64]*PathtestContext` can be something we can address once we see evidence it's an issue, what do you think?